src/app/modules/submission/receiver/receiver.utils.ts

```diff
@@ -113,7 +114,10 @@ export const addAttachmentToResponses = (
   if (isBodyVersion3AndAbove(body)) {
     Object.keys(body.responses).forEach((id) => {
       const response = body.responses[id] as ParsedClearFormFieldResponseV3
-      if (response.fieldType === BasicField.Attachment) {
+      if (
+        response.fieldType === BasicField.Attachment &&
+        checkIsAttachmentResponsesIdInMap(attachmentMap, id)
+      ) {
```

Comment:
Could we use this instead?

```suggestion
      if (response.fieldType === BasicField.Attachment && id in attachmentMap) {
```

If this serves the same purpose, then we might be able avoid changing the function shape of `isAttachmentResponseFromMap()`.

Reply:
hi @KenLSM, 

i've thought about this and the reason why I changed both if clauses to use the same `checkIsAttachmentResponsesIdInMap` function is so that the logic between both functions are consistent, with exception of the json schema (`response.answer.filename` for v3 vs `response.filename`). 

When i was debugging this issue, I was looking at the 2 if clauses for versions <3 and >=3 and trying to find out the difference, which I figured was just the json schema between the diff versions. 
Hence, i think changing both to use the same `checkIsAttachmentResponsesIdInMap` has the benefit of conveying this idea to future developers. 

As for renaming the function from `isAttachmentResponseFromMap` to `checkIsAttachmentResponsesIdInMap`, the rationale is to let function's names start with verbs for better readability. Hence, I took the opportunity to enforce some code style practices at the same time. 
From Oracle's style guide: <img width="1139" alt="image" src="https://github.com/user-attachments/assets/1adc99c6-f0de-4d74-a1d5-1873c3e4367f">

Ultimately, `id in attachmentMap` works as well and I'm good with using the above if desired, but i do believe the above approach has some of the above benefits mentioned.

Reply:
Alternatively, i can remove the function entirely and just do `id in attachmentMap` and `response._id in attachmentMap`

Reply:
Correct to note that functions should start with verbs, however for predicates (things that returns boolean that is meant to be used in if-else blocks), we don't use verbs.